libs/cardano-ledger-core/src/Cardano/Ledger/Credential.hs

```diff
@@ -118,22 +118,23 @@ instance Crypto c => FromJSONKey (Credential kr c) where
   fromJSONKey = FromJSONKeyTextParser parseCredential
     where
       parseCredential t = case T.splitOn "-" t of
-        ["scripthash", hash] ->
-          maybe
-            (badHex hash)
-            (pure . ScriptHashObj . ScriptHash)
-            (hashFromTextAsHex hash)
-        ["keyhash", hash] ->
-          maybe
-            (badHex hash)
-            (pure . KeyHashObj . KeyHash)
-            (hashFromTextAsHex hash)
+        [key, hash] -- for backwards compatibility we support with space and without
+          | key == "scriptHash" || key == "script hash" ->
```

Comment:
Actually, to keep it compatible it would have to be 
`| key == "scriptHash" || key == "scripthash" ->` 
 because `script hash`  wasn't the  *key* implementation, though it was the regular json implementation.
What a mess, I'm afraid I didn't make it better when I wrote this `fromJSONKey" version which is different from the  `fromJSON` - I had to chose between consistent but bad (with the space in it) or inconsistent but better. 
Anyway, glad you're cleaning it up and unifying it

Reply:
Ah, crap, I meant to do this for FromJSON. Thank you for catching it.

We don't really need to worry about backwards compatibility for FromJSONKey, since "keyhash" version was barely released and no-one really relies on it

Reply:
Fixed